src/panel/direction/DirectionPanel.jsx

```diff
@@ -92,6 +95,14 @@ export default class DirectionPanel extends React.Component {
     Promise.all(poiRestorePromises).then(([ origin, destination ]) => {
       persistentPointState.origin = origin;
       persistentPointState.destination = destination;
+
+      // Show the raw POI name/latlon while waiting for computeRoutes to complete
+      this.setState({
+        originInputText: origin ? origin.name : '',
+        destinationInputText: destination ? destination.name : '',
```

Comment:
To be consistent with the current implementation, addresses and streets should be rendered as labels (with city names)
```suggestion
        originInputText: origin ? origin.getInputValue() : '',
        destinationInputText: destination ? destination.getInputValue() : '',
```